metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/DropWizardMetricFactory.java

```diff
@@ -59,13 +59,13 @@ public TimeMetric timer(String name) {
     }
 
     @Override
-    public <T> Publisher<T> runPublishingTimerMetric(String name, Publisher<T> publisher) {
+    public <T> Publisher<T> decoratePublisherWithTimerMetric(String name, Publisher<T> publisher) {
         TimeMetric timer = timer(name);
         return Flux.from(publisher).doOnComplete(timer::stopAndPublish);
```

Comment:
I have a bug.

Write this test:

```
decoratePublisherWithTimerMetric(ANY, Mono.delay(100 ms).then(Mono.error(new RunTimeException())))
   .repeat(10);

assertThat(metric.executionTimingsFor(ANY)).hasSize(10);
```

(Maybe in a separate PR)

The current result will be one (if we are lucky, if we are not it fails). And it is a bug.